src/pybind/mgr/orchestrator/module.py

```diff
@@ -11,7 +11,7 @@
 from ceph.deployment.inventory import Device
 from ceph.deployment.drive_group import DriveGroupSpec, DeviceSelection, OSDMethod
 from ceph.deployment.service_spec import PlacementSpec, ServiceSpec, service_spec_allow_invalid_from_json, \
-    SNMPGatewaySpec
+    SNMPGatewaySpec, MDSSpec
```

Comment:
while I know SNMP is doing their import here, I think this should be taken from the ._interface file rather than the service_spec file directly to be in line with other specific service service spec imports. Honestly, I'd kind of like the SNMPGatewaySpec import to be from _interface as well.

Also, while we're talking about the _interface file, we should probably update the type signature of https://github.com/ceph/ceph/blob/master/src/pybind/mgr/orchestrator/_interface.py#L611 to take an mds spec the same way rgw, ingress etc. below it take their specific spec.

Reply:
Code has been updated accordingly. New code import SNMPGatewaySpec and MDSSpec from the interface.